### PR TITLE
Changes to build system to allow proper linking/control by JEDI system

### LINKED ISSUE #294: Allow JEDI control of UFS via NUOPC/ESMF
## Description
Some changes are required within UFS to allow for JEDI to link to UFS components and control the clock and state variables for data assimilation.

## Solution
Create individual libraries with exposed dependencies during the UFS build that can be linked into the JEDI data assimilation system. Set up build on NEMS side to allow for control from outside of NEMS (NEMS is no longer the top level driver). 

## Alternatives
So far, these changes are related to the build system, but there may be further modifications to source code required to enable JEDI to rewind the clock for the UFS and restart with updated state variables.

## Related to (coming shortly)
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/nems/issues/<issue_number>
- needed by noaa-emc/fv3atm/pull/<pr_number>

### PR DESCRIPTION
**Pull request by mark-a-potts**
## Description

This change exposes the linked libraries of sp and w3nco by setting them to PUBLIC so that they can be accessed by JEDI (or other system) when the UFS is linked in as a library.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)

- fixes ufs-community/ufs-weather-model/issues/294

## Testing

Testing on Hera is underway now.

## Dependencies

This is linked with a change in ufs-weather-model and NEMS.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,18 @@
 if(CCPP)
 
+  if(DEBUG)
```
Comment (climbfuji) [inline CMakeLists.txt:3]: This change (everything related to `_ccpp_debug_arg`) was merged into develop a few weeks back, see https://github.com/NOAA-EMC/fv3atm/blob/develop/CMakeLists.txt. Why is it showing up here again?

File: atmos_model.F90 [hunk 2]
```diff
@@ -100,7 +100,7 @@ module atmos_model_mod
 use IPD_driver,         only: IPD_initialize, IPD_initialize_rst
 use CCPP_driver,        only: CCPP_step, non_uniform_blocks
 
-use stochastic_physics_wrapper_mod, only: stochastic_physics_wrapper
+use stochastic_physics_wrapper_mod, only: stochastic_physics_wrapper,stochastic_physics_wrapper_end
```
Comment (climbfuji) [inline atmos_model.F90:103]: This as well.

**Review by climbfuji — CHANGES_REQUESTED** [review]
This PR shows a large amount of changes that are already in develop. It seems to me that the PR hasn't pulled in the latest code changes from develop.

File: CMakeLists.txt (hunk 1 above)
Comment (mark-a-potts) [reply CMakeLists.txt:3]: I don't know why the files show up here again, but when comparing directly to origin/develop using git, there is only a change to the CMakeLists.txt file that changes the ccppphysics link to PUBLIC.

**Comment by climbfuji** [comment]
This PR is included in and replaced by https://github.com/NOAA-EMC/fv3atm/pull/211.

**Pull request closed without merging**